api/v1beta1/machine_types.go

```diff
@@ -230,10 +230,32 @@ type MachineStatus struct {
 	// Conditions defines current service state of the Machine.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// Deletion contains information relating to removal of the Machine.
+	// Only present when the Machine has a deletionTimestamp and is being removed from the cluster.
+	// +optional
+	Deletion MachineStatusDeletion `json:"deletion,omitempty"`
```

Comment:
```suggestion
	Deletion *MachineStatusDeletion `json:"deletion,omitempty"`
```

I think in the past we used a pointer here so that this marshals to nothing instead of an empty object (cc @JoelSpeed WDYT?)

Reply:
It should be a pointer yes, optional structs need to be able to be omitted by the marshaller, so must be pointers as the json marshaller cannot determine if a struct reference is empty or not